pyro/distributions/one_hot_categorical.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+from torch.autograd import Variable
+
+from pyro.distributions.distribution import Distribution
+from pyro.distributions.util import get_probs_and_logits, torch_eye, torch_multinomial, torch_zeros_like
+
+
+class OneHotCategorical(Distribution):
+    """
+    OneHotCategorical (discrete) distribution.
+
+    Discrete distribution over one-hot vectors.
+
+    :param ps: Probabilities. These should be non-negative and normalized
+        along the rightmost axis.
+    :type ps: torch.autograd.Variable
+    :param logits: Log probability values. When exponentiated, these should
+        sum to 1 along the last axis. Either `ps` or `logits` should be
+        specified but not both.
+    :type logits: torch.autograd.Variable
+    :param batch_size: Optional number of elements in the batch used to
+        generate a sample. The batch dimension will be the leftmost dimension
+        in the generated sample.
+    :type batch_size: int
```

Comment:
can you add a `:returns :` to be explcit that it returns a one hot `Variable` of the same dims as `p`?